nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,467 @@
+    """
+    Prepare info for plotting one surface map on a single mesh.
+
+
+    This computes the dictionary that gets inserted in the web page,
+    which contains the encoded mesh, colors, min and max values, and
+    background color.
+
+    """
+    info = {}
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, surf_map, threshold, symmetric_cmap=symmetric_cmap)
+    info['inflated_left'] = to_plotly(surf_mesh)
+    info['vertexcolor_left'] = _get_vertexcolor(
+        surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = False
+    info['colorscale'] = colors
+    return info
+
+
+def _check_mesh(mesh):
+    if not isinstance(mesh, str):
+        assert isinstance(mesh, collections.Mapping)
+        assert {'pial_left', 'pial_right', 'sulc_left', 'sulc_right',
+                'infl_left', 'infl_right'}.issubset(mesh.keys())
+        return mesh
+    mesh = datasets.fetch_surf_fsaverage(mesh)
+    return mesh
+
+
+def full_brain_info(volume_img, mesh='fsaverage5', threshold=None,
+                    cmap=cm.cold_hot, black_bg=False, symmetric_cmap=True,
+                    vol_to_surf_kwargs={}):
+    """
+    Project 3d map on cortex; prepare info to plot both hemispheres.
+
+
+    This computes the dictionary that gets inserted in the web page,
+    which contains encoded meshes, colors, min and max values, and
+    background color.
+
+    """
+    info = {}
+    mesh = _check_mesh(mesh)
+    surface_maps = {
+        h: surface.vol_to_surf(volume_img, mesh['pial_{}'.format(h)],
+                               **vol_to_surf_kwargs)
+        for h in ['left', 'right']
+    }
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, np.asarray(list(surface_maps.values())).ravel(), threshold,
+        symmetric_cmap=symmetric_cmap)
+
+    for hemi, surf_map in surface_maps.items():
+        bg_map = surface.load_surf_data(mesh['sulc_{}'.format(hemi)])
+        info['pial_{}'.format(hemi)] = to_plotly(mesh['pial_{}'.format(hemi)])
+        info['inflated_{}'.format(hemi)] = to_plotly(
+            mesh['infl_{}'.format(hemi)])
+
+        info['vertexcolor_{}'.format(hemi)] = _get_vertexcolor(
+            surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = True
+    info['colorscale'] = colors
+    return info
+
+
+def _fill_html_template(info, embed_js=True):
+    as_json = json.dumps(info)
+    as_html = get_html_template('surface_plot_template.html').replace(
+        'INSERT_STAT_MAP_JSON_HERE', as_json)
+    as_html = add_js_lib(as_html, embed_js=embed_js)
+    return SurfaceView(as_html)
+
+
+def view_img_on_surf(stat_map_img, mesh='fsaverage5',
+                     threshold=None, cmap=cm.cold_hot,
```

Comment:
and symmetric_cbar ?

Reply:
> Any reason why 'interpolation' and 'kind' are not exposed ?

to avoid cluttering the interface. they can easily be added if you think it is better. @GaelVaroquaux  what do you think?

Reply:
> and symmetric_cbar ?

do you mean symmetric_cbar or symmetric_cmap?

- if symmetric_cmap, because it is mostly useful for plotting atlases, but this is only relevant when providing a surface atlas, i.e. using view_surf. projection of volume atlases on the surface usually don't look good. But I guess we could add it for plotting statistical maps that have only positive or only negative values; I don't know if this is a real use case?

- if symmetric_cbar, we don't provide this option here, only in plot_stat_map etc. if the colormap is symmetric the colorbar will be too. I could add it but I'm not sure it is super useful